### PR TITLE
Linear Combination WC Scans

### PR DESCRIPTION
**Pull request by sunnyliu1220**
- Add scans over linear combinations of WCs.
- Consolidate all validation notebooks into one.
- Build all preprocessing layers into the DNN.
- NOT backward compatible.

### CODE REVIEW

File: training/README.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 This folder contains current NN training code. Make sure to include all files to be transferred in `train.submit`! Also, check the graph and model saving lines in `train.py` if you want to convert it to a notebook or vice versa, otherwise they might not save correctly.
 
-Note: The current training set is at `/scratch365/klannon/dnnlikelihood/likelihood_data_no_delta.npz`.
\ No newline at end of file
+Note: The current training set is at `/scratch365/klannon/dnnlikelihood/likelihood_data_no_delta.npz`. A copy can also be found on CurateND.
```
Comment (NJamieson22) [inline training/README.md:5]: idk how CurateND works but perhaps add a link to where it is?

**Review by NJamieson22 — APPROVED** [review]
Everything looks good to me. I didn't look too much at the new linear combination 2D validation stuff yet (as I do not quite understand it as of now), but everything else seems right to me.

File: training/README.md (hunk 1 above)
Comment (sunnyliu1220) [reply training/README.md:5]: Done

File: validation/README.md [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
-- `validation.ipynb`: 1D validation without self-consistent profiling. This means the profiled scans merely evaluate the NN at the same points in 16D space as the `Combine` scans.
-- `validation_2d.ipynb`: 2D validation without self-consistent profiling.
-- `validation_2d_profiling.ipynb`: 2D validation with self-consistent profiling. The same as `2d_profiling.py`, just in a notebook.
-- `comparison_graphs`: Compares 1D validation without self-consistent profiling accuracy of two models.
-- `graphs\`: Stores the graphs for the validation.
+- `validation_all.sh`, `validation_all.submit`, and `validation_all.py` are to be used together to perform all validation, including self-consistent profiling, which means that for each value of the WC being scanned, the other 15 WCs are fit to produce the minimum $2\Delta NLL$.
+- `validation_all.ipynb`: All validation. Has the same functionality as `validation_all.py`, just in a notebook.
+- `comparison_plots.ipynb`: Compares 1D validation without self-consistent profiling accuracy of two models. Needs to be updated.
```
[the remark this answers is not in the document]
Comment (sunnyliu1220) [reply validation/README.md:6]: Sounds good!

**Pull request merged**